### PR TITLE
feat(navbar): Add title area in the middle

### PR DESCRIPTION
**Pull request by astaev**
Closes #  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [x] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [x] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/navbar/navbar.component.ts [hunk 1]
```diff
@@ -17,9 +17,15 @@ import { IgxIconModule } from '../icon/public_api';
  */
 @Directive({
     // tslint:disable-next-line:directive-selector
-    selector: 'igx-action-icon'
+    selector: '[igxNavbarAction]'
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/navbar/navbar.component.ts:20]:
```suggestion
    selector: 'igx-navbar-action,[igxNavbarAction]'
```

File: projects/igniteui-angular/src/lib/navbar/navbar.component.ts [hunk 2]
```diff
@@ -17,9 +17,15 @@ import { IgxIconModule } from '../icon/public_api';
...
 })
-export class IgxActionIconDirective { }
+export class IgxNavbarActionDirective { }
+
+@Directive({
+    selector: '[igxNavbarTitle]',
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/navbar/navbar.component.ts:25]:
```suggestion
    selector: 'igx-navbar-title,[igxNavbarTitle]',
```

File: projects/igniteui-angular/src/lib/navbar/navbar.component.html [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
-        <igx-icon (click)="_triggerAction()" fontSet="material" *ngIf="isActionButtonVisible">{{actionButtonIcon}}</igx-icon>
-        <ng-content select="igx-action-icon"></ng-content>
-        <h1 class="igx-navbar__title" [attr.id]="titleId">{{ title }}</h1>
+        <igx-icon (click)="_triggerAction()" fontSet="material" *ngIf="isActionButtonVisible">{{actionButtonIcon}}
+        </igx-icon>
+        <ng-content select="[igxNavbarAction]"></ng-content>
+        <h1 *ngIf="!isTitleContentVisible" class="igx-navbar__title" [attr.id]="titleId">{{ title }}</h1>
+    </div>
+    <div class="igx-navbar__middle">
+        <ng-content select="[igxNavbarTitle]"></ng-content>
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/navbar/navbar.component.html:9]: We don't want a middle area. Move the igxNavbarTitle directive next to the `h1` title.

File: projects/igniteui-angular/src/lib/navbar/navbar.component.ts (hunk 1 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/navbar/navbar.component.ts]: Done

File: projects/igniteui-angular/src/lib/navbar/navbar.component.ts (hunk 2 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/navbar/navbar.component.ts]: Done

File: projects/igniteui-angular/src/lib/navbar/navbar.component.html (hunk 3 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/navbar/navbar.component.html]: Done

**Pull request closed without merging**